src/Api/CommandHandler/AddGiftCardToOrderHandler.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGiftCardPlugin\Api\CommandHandler;
+
+use Setono\SyliusGiftCardPlugin\Api\Command\AddGiftCardToOrder;
+use Setono\SyliusGiftCardPlugin\Applicator\GiftCardApplicatorInterface;
+use Setono\SyliusGiftCardPlugin\Model\GiftCardInterface;
+use Setono\SyliusGiftCardPlugin\Model\OrderInterface;
+use Setono\SyliusGiftCardPlugin\Repository\GiftCardRepositoryInterface;
+use Setono\SyliusGiftCardPlugin\Repository\OrderRepositoryInterface;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Webmozart\Assert\Assert;
+
+final class AddGiftCardToOrderHandler implements MessageHandlerInterface
+{
+    private GiftCardRepositoryInterface $giftCardRepository;
+
+    private OrderRepositoryInterface $orderRepository;
+
+    private GiftCardApplicatorInterface $giftCardApplicator;
+
+    public function __construct(
+        GiftCardRepositoryInterface $giftCardRepository,
+        OrderRepositoryInterface $orderRepository,
+        GiftCardApplicatorInterface $giftCardApplicator
+    ) {
+        $this->giftCardRepository = $giftCardRepository;
+        $this->orderRepository = $orderRepository;
+        $this->giftCardApplicator = $giftCardApplicator;
+    }
+
+    public function __invoke(AddGiftCardToOrder $command): GiftCardInterface
+    {
+        $giftCardCode = $command->getGiftCardCode();
+        Assert::notNull($giftCardCode);
+
+        $giftCard = $this->giftCardRepository->findOneByCode($giftCardCode);
+        Assert::notNull($giftCard);
+
+        /** @var OrderInterface|null $order */
+        $order = $this->orderRepository->findOneBy(['tokenValue' => $command->orderTokenValue]);
+        Assert::notNull($order);
+
+        $this->giftCardApplicator->apply($order, $giftCard);
+
+        return $giftCard;
```

Comment:
Is there a reason why we return the gift card? Is that how API platform always does it?

Reply:
That's how Sylius does. They apparently return the object for each Handler (see AddressOrderHandler, ChangePaymentMethodHandler, ...)